### ISSUE TITLE
collection.insert() is not updating the redis subscription (no event "i" received by subscription manager)

### ISSUE THREAD

**Issue by rj-david** [cult-of-coders/redis-oplog#265]
Meteor version: 1.6.1
Redis Oplog version: 1.2.7
Mongo DB version: 3.4.10
Redis version: 4.0.8

Problem: Inserting a new record in a collection does not update the subscription

Representative code:
`Notifications.insert(data, {
  namespace: 'userId::' + userId
});
`

But updating the records work. With debug on, I can see that the subscription manager has no event "i" when inserting records.
```
I20180328-08:00:34.303(8)? [1522195234303] - [RedisSubscriptionManager] Subscribing to channel: userId::ccCN3JoGPSiWYyTf8::notifications
I20180328-08:00:55.231(8)? [1522195255231] - [RedisSubscriptionManager] Received event: "u" to "userId::ccCN3JoGPSiWYyTf8::notifications"
```

As of now, my simple remedy is to immediately update the record after inserting it so that it will update the subscription.

**Comment by theodorDiaconu** [comment]
@rj-david quick reproduction repo please? Thanks!

**Comment by rj-david** [comment]
@theodorDiaconu 

Sorry, it took some time.

https://github.com/rj-david/MeteorRedisOplogTest

If you go to the file "imports/api/collections/Notifications.js", you will see there the function newNotification()

In that function, I need to call notificationOn which is an update function just to update the subscription.

The publication is in the following files:
imports/api/publications/test.js
imports/api/helpers/validatedPublication.js

In package.json, please note that my ports are not the default ports of mongodb and redis

Thanks in advance.

**Comment by theodorDiaconu** [comment]
Thank you @rj-david I will book this week time to sweep all the bugs from it!

**Comment by nathan-muir** [comment]
@rj-david had a quick look at your repo.

The issue is related to `aldeed:collection2` / `aldeed:collection2-core` interacting with `cultofcoders:redis-oplog`

---
**TLDR** _quickfix_ put `aldeed:collection2` at the TOP of your `.meteor/packages` (above `redis-oplog`).

OR, slightly more robust:

```js
Notifications.configureRedisOplog({
  mutation(options, {event, selector, modifier, doc}) { 
     if (event === 'i' && doc.userId != null) {
         options.namespace = `userId::${doc.userId}`
     } else if ((event === 'u' || event === 'r') && selector.userId != null) {
         options.namespace = `userId::${selector.userId}`  
     }
  }
 cursor(options, selector) {
   if (selector.userId != null) {
      options.namespace = `userId::${selector.userId}`
   }
 }
})
```
----

**Explanation**

[`aldeed:collection2` does _not_ pass options from `Collection.insert(doc, options)` down to the `_overriden.insert`](https://github.com/aldeed/meteor-collection2/blob/master/package/collection2/collection2.js#L209-L210)

So when you perform `Notifications.insert(doc, {namespace: 'userId::' + userId})` the `namespace` is not actually reaching redis-oplog.

Note, that [`redis-oplog` doesn't pass the options down either](https://github.com/cult-of-coders/redis-oplog/blob/master/lib/mongo/Mutator.js#L43) so the easy solution (putting `aldeed:collection2` first) means that you can't pass options to collection2 like `Collection.insert(doc, {validate: false})`...

So probably better to go with `.configureRedisOplog` on the collection....

**Comment by rj-david** [comment]
Nice catch, @nathan-muir. I'll do that once I got home.

**Comment by rj-david** [comment]
@nathan-muir, it worked. And I also learned something new. This is a lot better than individually adding the namespace/channel for each query. Love it. Thanks @theodorDiaconu  and @nathan-muir
